packages/rainbowkit/src/transactions/TransactionStoreContext.tsx

```diff
@@ -39,6 +56,12 @@ export function TransactionStoreProvider({
     }
   }, [store, address, chainId]);
 
+  useEffect(() => {
+    if (store && address && chainId) {
+      return store.onTransactionStatus(onTransactionStatus);
+    }
+  }, [store, address, chainId, onTransactionStatus]);
```

Comment:
We could pass this logic in [here](https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/transactions/useRecentTransactions.ts#L20-L22) which would be less work, but that would cause a lot of re-renders + make unnecessary RPC calls since it'll happen pretty often due to how our transaction store provider works. 

We'll call refetch whenever is necessary to update our balance (e.g when a tx has successfully passed).